### PR TITLE
test: automate regex tests

### LINKED ISSUE #125: Make unit tests and `gen_log_guide.ts` pull example log lines from a common resource
### Feature(s) Requested

Per @valarnin comment on #121 :

> the unit tests and the gen_log_guide.ts files should pull from a common resource somewhere, rather than duplicating these lines

### PR DESCRIPTION
**Pull request by wexxlee**
Closes #125 - last of the 3 PRs.

This is based on the changes in ~137 (`1fa2945`) and~ (Edit: 137 merged & rebased out) 138 (`3275db3`), so that commit can be ignored here & will be rebased out when the earlier PRs are merged.

### CODE REVIEW

**Review by valarnin — COMMENTED** [review]
You seem to have decoupled the unit tests and line examples here, but not in #138.

Also, you mention 0x16 lines:
```
// example_log_lines doesn't have specific examples of 0x16 lines, but they should be matched
```

Should we throw an issue up to add these examples after this PR and #138 are merged?

---

Aside from the comment below and the issues with changes on this vs #138, this looks good to me. I'll wait to approve until after 138 is merged and this is rebased.

**Comment by wexxlee** [comment]
> You seem to have decoupled the unit tests and line examples here, but not in #138.

Intentional - see original comment on #138 (just replied there as well).  Might make more sense at this point to just close #138 and have this PR handle all the changes, since it already includes all of #138 in `3275db3`.

> Also, you mention 0x16 lines:
> 
> ```
> // example_log_lines doesn't have specific examples of 0x16 lines, but they should be matched
> ```
> 
> Should we throw an issue up to add these examples after this PR and #138 are merged?

Yeah, good idea.  #158 added.

File: test/helper/example_log_lines_test_data.ts
Comment (wexxlee) [reply test/helper/example_log_lines_test_data.ts:37]: > You could potentially reference the exact line here instead of using an index, now that they're decoupled? Same thing in #138 once the tests and examples are decoupled there as well.

I think that's a good change.  Let me know your thoughts on how you want to handle merging #138/#139 (see comment on #138), and I'll figure out whether to make that change here or there.

**Comment by valarnin** [comment]
> Intentional - see original comment on #138 (just replied there as well). Might make more sense at this point to just close #138 and have this PR handle all the changes, since it already includes all of #138 in `3275db3`.

Yeah, I'm fine with just closing 138 and including everything as part of this PR instead. The actual scope of this PR is significantly smaller than the line change count in github suggests, since it's more a lot of moving things around rather than lines actually being added/deleted in a more traditional sense.

**Comment by wexxlee** [comment]
OK, I think this should be ready now.

**Review by valarnin — APPROVED** [review]
LGTM.

Sorry for the delay, I wanted to ensure this worked as expected locally but apparently messed up my node install in my WSL instance and had to rebuild the WSL instance to fix it.

**Pull request merged**
